File: lab-2.3/c#/Widow.cs
namespace CharacterLib
{
    public class Widow : IDescribable
    {
        private string name;
        private int skillLevel;

        public Widow()
        {
            name = "Unknown";
            skillLevel = 60;
        }

        public Widow(string name, int skillLevel)
        {
            this.name = name;
            this.skillLevel = skillLevel;
        }

        public Widow(Widow other)
        {
            name = other.name;
            skillLevel = other.skillLevel;
        }

        public string GetName()
        {
            return name;
        }

        public int GetPower()
        {
            return skillLevel;
        }

        public string Describe()
        {
            return $"Black Widow: {name}, Skill: {skillLevel}";
        }
    }
}

File: lab-2.3/c++/Widow.cpp
#include "Widow.h"

Widow::Widow() : name("Unknown"), skillLevel(60) {}

Widow::Widow(std::string n, int sl)
    : name(n), skillLevel(sl) {}

Widow::Widow(const Widow& other)
    : name(other.name), skillLevel(other.skillLevel) {}

std::string Widow::getName() const {
    return name;
}

int Widow::getPower() const {
    return skillLevel;
}

std::string Widow::describe() const {
    return "Black Widow: " + name + ", Skill: " + std::to_string(skillLevel);
}

IDescribable* Widow::clone() const {
    return new Widow(*this);
}